src/client/menu/menu.c

```diff
@@ -2437,572 +2437,572 @@ static const char *idcredits[] = {
+				n ++;
+				break;
+			}
+		}
+
+		creditsIndex[n] = 0;
+		credits = (const char **)creditsIndex;
+	}
+	else
+	{
+		if (M_IsGame("xatrix")) /* Xatrix - The Reckoning */
+		{
+			credits = xatcredits;
+		}
+
+		else if (M_IsGame("rogue")) /* Rogue - Ground Zero */
+		{
+			credits = roguecredits;
+		}
+
+		else
+		{
+			credits = idcredits;
+		}
+	}
+
+	credits_start_time = cls.realtime;
+	M_PushMenu(M_Credits_MenuDraw, M_Credits_Key);
+}
+
+/*
+ * MODS MENU
+ */
+
+static menuframework_s s_mods_menu;
+static menulist_s s_mods_list;
+static menuaction_s s_mods_apply_action;
+static char mods_statusbar[64];
+
+static char **modnames = NULL;
+static int nummods;
+
+static void
+Mods_NamesInit(void)
+{
+	/* initialize list of mods once, reuse it afterwards (=> it isn't freed) */
+	if (modnames == NULL)
+	{
+		modnames = FS_ListMods(&nummods);
+	}
+}
+
+static void
+ModsListFunc(void *unused)
+{
+	if (strcmp(BASEDIRNAME, modnames[s_mods_list.curvalue]) == 0)
+	{
+		strcpy(mods_statusbar, "Quake II");
+	}
+	else if (strcmp("ctf", modnames[s_mods_list.curvalue]) == 0)
+	{
+		strcpy(mods_statusbar, "Quake II Capture The Flag");
+	}
+	else if (strcmp("rogue", modnames[s_mods_list.curvalue]) == 0)
+	{
+		strcpy(mods_statusbar, "Quake II Mission Pack: Ground Zero");
+	}
+	else if (strcmp("xatrix", modnames[s_mods_list.curvalue]) == 0)
+	{
+		strcpy(mods_statusbar, "Quake II Mission Pack: The Reckoning");
+	}
+	else
+	{
+		strcpy(mods_statusbar, "\0");
+	}
+}
+
+static void
+ModsApplyActionFunc(void *unused)
+{
+	if (!M_IsGame(modnames[s_mods_list.curvalue]))
+	{
+		if(Com_ServerState())
+		{
+			// equivalent to "killserver" cmd, but avoids cvar latching below
+			SV_Shutdown("Server is changing games.\n", false);
+			NET_Config(false);
+		}
+
+		// called via command buffer so that any running server has time to shutdown
+		Cbuf_AddText(va("game %s\n", modnames[s_mods_list.curvalue]));
+
+		// start the demo cycle in the new game directory
+		menu_startdemoloop = true;
+
+		M_ForceMenuOff();
+	}
+}
 
 static void
 Mods_MenuInit(void)
 {
-    int currentmod;
-    int x = 0;
-    int y = 0;
-    char modname[MAX_QPATH]; //TG626
+	int currentmod;
+	int x = 0;
+	int y = 0;
+	char modname[MAX_QPATH]; //TG626
 
-    Mods_NamesInit();
+	Mods_NamesInit();
 	
-    // create array of bracketed display names from folder names - TG626
-    char **displaynames=malloc(sizeof(*displaynames) * nummods);
-
-    for (int i=0; i < nummods; i++)
-    {
-        strcpy(modname, "[");
-	if (strlen(modnames[i]) < 16)
-        {
-	    strcat(modname, modnames[i]);
-	    for (int j=0; j < 15 - strlen(modnames[i]); j++)
-	    {
-	        strcat(modname, " ");
-	    }
-        }
-        else
-        {
-            strncat(modname, modnames[i], 12);
-            strcat(modname, "...");
-        }	
-        strcat(modname, "]");
+	// create array of bracketed display names from folder names - TG626
+	char **displaynames=malloc(sizeof(*displaynames) * nummods);
```

Comment:
this should've been (nummods+1), otherwise `displaynames[nummods] = 0;` below acccesses invalid memory
(just a heads-up, I already fixed this)